beacon-chain/state/state-native/state_trie.go

```diff
@@ -85,14 +85,14 @@ var (
 		types.HistoricalSummaries,
 	}
 
-	capellaFields = append(
-		append([]types.FieldIndex{}, altairFields...),
-		append([]types.FieldIndex{types.LatestExecutionPayloadHeaderCapella}, withdrawalAndHistoricalSummaryFields...)...,
+	capellaFields = slices.Concat(
+		altairFields,
+		append([]types.FieldIndex{types.LatestExecutionPayloadHeaderCapella}, withdrawalAndHistoricalSummaryFields...),
 	)
 
-	denebFields = append(
-		append([]types.FieldIndex{}, altairFields...),
-		append([]types.FieldIndex{types.LatestExecutionPayloadHeaderDeneb}, withdrawalAndHistoricalSummaryFields...)...,
+	denebFields = slices.Concat(
+		altairFields,
+		append([]types.FieldIndex{types.LatestExecutionPayloadHeaderDeneb}, withdrawalAndHistoricalSummaryFields...),
```

Comment:
concat takes several inputs, this can be written as 
```
    denebFields = slices.Concat(
	altairFields,
	[]types.FieldIndex{types.LatestExecutionPayloadHeaderDeneb},
	withdrawalAndHistoricalSummaryFields,
)
```